helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -657,6 +678,68 @@ public void enableP2PMessage(boolean enabled) {
     _record.setBooleanField(HelixConfigProperty.P2P_MESSAGE_ENABLED.name(), enabled);
   }
 
+  /**
+   * Set the required Instance Capacity Keys.
+   * @param capacityKeys
+   */
+  public void setInstanceCapacityKeys(List<String> capacityKeys) {
+    if (capacityKeys == null || capacityKeys.isEmpty()) {
```

Comment:
For those public methods and the throw exception patterns in several methods, I have a concern. Because Helix library gets exposed and these changes will become something cannot be changed in backwards. To be extra safe, do you know who'll be the client calling the public methods? If it's only helix controller itself, there's some confidence level of data being passed in, probably we can avoid some throw statements.

Reply:
The way a user can provide capacity information is:
1. get the config object using Zk accessor.
2. Update the config object, using this call.
3. update the config to Zk through Zk accessor.

So we need this interface. And we need to fail the error config early before they reach the ZK. That's why throw exception is necessary.